### DISCUSSION TITLE
[Feature] Automatic Panoramic detection and creation

### DISCUSSION THREAD

**Discussion by Underscan007** [immich-app/immich#14517] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

The built-in panoramic feature on mobile phones often produces low-quality results. To achieve better panoramas, I take multiple pictures with either my DSLR or smartphone and then stitch them together using a dedicated mobile app (e.g., Bimostitch Pro). However, this process is somewhat tedious: first, I capture the photos, then transfer them to the panoramic app, and finally move the resulting panorama back to the appropriate folder for automatic upload to Immich.

During this workflow, the link between the original photos and the stitched panorama is lost.

It would be fantastic if Immich could incorporate a panoramic detection and stitching feature directly into its processing pipeline. This could work by identifying photos taken in quick succession and performing image analysis to determine potential panoramas.

Such a feature could complement the planned "auto-stacking" functionality

### Platform

- [X] Server
- [X] Web
- [ ] Mobile

**Comment by bo0tzz** [comment]
#4572
